### DISCUSSION TITLE
Vladmandic Test Drive by Automatic1111 user

### DISCUSSION THREAD

**Discussion by CalculonPrime** [vladmandic/sdnext#3564] [Q&A] [answered]
Installed Vladmandic tonight to see how it compares to Automatic 1111 which I'm used to.  Some questions/comments:

- Is there a light theme available?  How do I switch themes?
- How can I start the web GUI without updating and/or contacting the internet?
- Can you add support for soft links in the models directory hierarchy?  Automatic 1111 supports this.  I soft-linked in a model file, and a whole directory of models from a central location (shared by different apps) and Vladmandic didn't detect the files/directories.  I had to add the model file as a hard link in order to get it to work.
- I think you need to add some space between the number and the ^v  value selectors on the generate from text page (and probably elsewhere).  For me, the number is jammed up against the arrow icons with no space between the two and its hard to read.

Thanks.

System: Windows 10, Firefox 128ESR

**Comment by vladmandic** [comment]
- settings -> user interface
- there is no update unless you specify `--update`. but initial install needs to install libraries, so it needs to be online
- don't link files. soft-link models folder. or specify path in settings -> paths. or specify models root for all model types, not just diffusion models.
- i'll take a look at css for that

**Reply by CalculonPrime** [reply]
- OK, found the theme settings. I see two light themes, but using them is difficult because the background behind text (in the picker dropdowns for example) is too dark.  The issue is worst with "simple light".
- When I run with no options, after installation, my firewall alerts me to python attempting to contact api.github.com
- OK, but I'd prefer finer-grain control over exactly what models are made available to different installed web UIs.
- Thanks

**Reply by vladmandic** [reply]
not many ppl use light themes. if you want to contribute one, go for it - theme is just a css

regarding git, it's a version check, not upgrade.
you can disable it with cmd flag.

regarding models, perhaps you could symlink subfolders?

**Reply by CalculonPrime** [reply]
I tried symlinking the "Stable Diffusion" folder under models, but it didn't seem to work.

I guess I can just hardlink models to solve this, but it is slightly harder to track what's going on (if I forget what I've done later), and you can't hardlink between volumes, which I what I do for several models in automatic1111's web UI.

If you wanted to support symlinks, I think you'd just need to tweak the flags or filtering of directories when you scan them for models.

**Reply by vladmandic** [reply]
you cannot symlink stable-diffusion folder itself!
you can rehost models folder, you can rehost stable-diffusion folder or you can symlink subfolders of it.

you can also specify any of the above using command line params so you don't need to go into settings, etc.

**Reply by CalculonPrime** [reply]
Are any config files stored in the models directory hierarchy that could potentially conflict between different web UIs, if I decided to share the entire models folder between them?  For example, I think automatic 1111's web UI stores images in the LORA subdirectory associated with each LORA that show up as the thumbnail image.  Or is that part of Stable Diffusion itself?

**Reply by vladmandic** [reply, marked as answer]
no conflicts. any config files that are in models/stable-diffusion or models/lora are purely informational (e.g. lora metadata downloaded from civitai) and do not impact model behavior.
regarding images, its pretty standard to have image next to model (actual model or lora or whatever) as preview, sdnext supports that as well.

**Comment by CalculonPrime** [comment]
When I start the web UI with "launch.py --skip-git" then the initial internet access to api.github.com is avoided, but then when I open the local URL in the browser, I still get an attempted access to vladmandic.github.io.  When I block it with my firewall the command shell reports:
```
21:34:06-451760 ERROR    MOTD: HTTPSConnectionPool(host='vladmandic.github.io', port=443): Max retries exceeded with
                         url: /automatic/motd (Caused by NewConnectionError('<urllib3.connection.HTTPSConnection object
                         at 0x0000029912747710>: Failed to establish a new connection: [WinError 10013] An attempt was
                         made to access a socket in a way forbidden by its access permissions'))
```

**Reply by vladmandic** [reply]
you can disable MOTD in settings (message of the day). actually it's not used at all right now, I'll switch it to disabled by default.

**Comment by CalculonPrime** [comment]
Thanks for answering these questions.  Should I add additional issues I face to this thread or start new discussions?

**Reply by vladmandic** [reply]
small ones, add here.
anything larger, better to have a dedicated thread.
